### DISCUSSION TITLE
Problem with setting up upstream DNS and PRT to show hostnames

### DISCUSSION THREAD

**Discussion by duka10** [AdguardTeam/AdGuardHome#5570] [Q&A]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not ask a question

### Operating system type

FreeBSD

### CPU architecture

AMD64

### Installation

GitHub releases or script from README

### Setup

On a router, DHCP is handled by the router

### AdGuard Home version

0.107.25

### Description

**#### What did you do?**

First time with AgGuard. I used Pi hole. I install it on OPNsense. 
Everything working great, but i tried to set up ptr and upstream DNS to have a proper hostnames od network devices but i stuck there.

I set Upstream DNS servers and Private reverse DNS servers.

For upsteam i tried [/168.192.in-addr.arpa/]192.168.1.1,  [/lan/]192.168.1.1 but no luck.

Private reverse DNS server is set to 192.168.1.1

**#### Actual result**

Still no hostnames are shown, only IP addresses.

**#### Additional information**

### **I try with dig and this is what i got.**

duka@ubuntu:~$ dig -x 192.168.1.11 @192.168.1.1

; <<>> DiG 9.18.1-1ubuntu1.2-Ubuntu <<>> -x 192.168.1.11 @192.168.1.1
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 45874
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 0

;; QUESTION SECTION:
;11.1.168.192.in-addr.arpa.     IN      PTR

;; Query time: 4 msec
;; SERVER: 192.168.1.1#53(192.168.1.1) (UDP)
;; WHEN: Fri Mar 03 18:26:35 UTC 2023
;; MSG SIZE  rcvd: 43

### **I set up logs and got this results:**

2023/03/03 19:06:24.312632 26789#36 [debug] rdns: "192.168.1.11" added to queue
2023/03/03 19:06:24.312656 26789#36 [debug] rdns: "192.168.1.15" added to queue
2023/03/03 19:06:24.312678 26789#36 [debug] rdns: "192.168.1.16" added to queue
2023/03/03 19:06:24.312687 26789#34 [debug] dnsproxy: cache: disabled; not caching
2023/03/03 19:06:24.312696 26789#36 [debug] rdns: "192.168.1.18" added to queue
2023/03/03 19:06:24.312725 26789#34 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).replyFromUpstream(): RTT: 2.287µs
2023/03/03 19:06:24.312726 26789#36 [debug] rdns: "192.168.20.10" added to queue
2023/03/03 19:06:24.312748 26789#36 [debug] rdns: "192.168.30.10" added to queue
2023/03/03 19:06:24.312753 26789#34 [debug] rdns: resolving "192.168.1.11": all upstreams failed to exchange request
2023/03/03 19:06:24.312765 26789#36 [debug] rdns: "192.168.1.17" added to queue
2023/03/03 19:06:24.312787 26789#34 [debug] dnsproxy: cache: disabled; not caching
2023/03/03 19:06:24.312790 26789#36 [debug] rdns: "192.168.1.100" added to queue
2023/03/03 19:06:24.312810 26789#34 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).replyFromUpstream(): RTT: 690ns
2023/03/03 19:06:24.312824 26789#34 [debug] rdns: resolving "192.168.1.15": all upstreams failed to exchange request
2023/03/03 19:06:24.312853 26789#34 [debug] dnsproxy: cache: disabled; not caching
2023/03/03 19:06:24.312875 26789#34 [debug] github.com/AdguardTeam/dnsproxy/proxy.(*Proxy).replyFromUpstream(): RTT: 649ns
2023/03/03 19:06:24.313007 26789#34 [debug] rdns: resolving "192.168.20.10": all upstreams failed to exchange request

**Comment by ainar-g** [comment]
Hello.  I'm not sure about your network configuration.  Is `192.168.1.1` the address of AdGuard Home or the address of your router's own DNS server?  Because if you want the rDNS to work with the router's DHCP hosts, you should put the address/es of the router's own DNS server to the *Private reverse DNS servers* input.  Otherwise, you're just routing those to AGH itself.

**Reply by duka10** [reply]
192.168.1.1 is address of Opnsense router and Adguard is installed as plugin. DHCP is set on router and DNS server is set to Adguard.
In Private reverse DNS servers i set 192.168.1.1 address.

And also i set Upstream DNS servers.

For upsteam i tried [/168.192.in-addr.arpa/]192.168.1.1, [/lan/]192.168.1.1 but no luck.


![Pict](https://user-images.githubusercontent.com/93546291/223220794-cf360264-2988-4ce1-85e4-e25e702016de.jpg)

**Reply by ainar-g** [reply]
If that is the case then it seems like you are routing AdGuard Home to itself.  You should probably consult the plugin's documentation, but you basically need to put the address of the router's DNS server into the *Private reverse DNS servers* input, not of AdGuard Home itself.

**Reply by duka10** [reply]
I have same ip address for Opnsense and AdGuard, its just different port. I will try put Unbound DNS from Opnsense router in Private reverse DNS servers.

**Comment by duka10** [comment]
Hi. I fugured how to solve this and i finally got hostnames. But now in every hostname i have included domain for my OPNsense router.

Is it possible to exclude that? In attachment is part of my clients list in AdGuard. Thanks.
![Screenshot 2023-09-18 190829](https://github.com/AdguardTeam/AdGuardHome/assets/93546291/713183b2-8a18-4354-bc48-a7db8a650dd0)

**Reply by quincy5891** [reply]
how did you get it to work?

**Reply by dhanadhan** [reply]
Please let us know how?
